pkg/gui/commits_panel.go

```diff
@@ -8,172 +8,395 @@ import (
+		return err
+	}
+
+	err = gui.handleCommitSelect()
+	if err != nil {
+		gui.Log.Errorf("Failed to handleCommitSelect at handleCommitSquashDown: %s\n", err)
+		return err
 	}
-	return false
+
+	return nil
 }
 
+// handleCommitFixup is called when a user wants to fix a commit.
+// g and v are passed to by the gocui library but only v is used.
+// If anything goes wrong it returns an error.
 func (gui *Gui) handleCommitFixup(g *gocui.Gui, v *gocui.View) error {
+
 	if len(gui.State.Commits) == 1 {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("YouNoCommitsToSquash"))
+
+		err := gui.createErrorPanel(gui.Tr.SLocalize("YouNoCommitsToSquash"))
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleCommitFixup: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
+
 	if gui.anyUnStagedChanges(gui.State.Files) {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("CantFixupWhileUnstagedChanges"))
+		err := gui.createErrorPanel(gui.Tr.SLocalize("CantFixupWhileUnstagedChanges"))
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleCommitFixup: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
+
 	branch := gui.State.Branches[0]
-	commit, err := gui.getSelectedCommit(g)
+
+	commit, err := gui.getSelectedCommit()
 	if err != nil {
+		gui.Log.Errorf("Failed to get selected commit: %s\n", err)
 		return err
 	}
+
 	message := gui.Tr.SLocalize("SureFixupThisCommit")
-	gui.createConfirmationPanel(g, v, gui.Tr.SLocalize("Fixup"), message, func(g *gocui.Gui, v *gocui.View) error {
-		if err := gui.GitCommand.SquashFixupCommit(branch.Name, commit.Sha); err != nil {
-			return gui.createErrorPanel(g, err.Error())
-		}
-		if err := gui.refreshCommits(g); err != nil {
-			panic(err)
-		}
-		return gui.refreshStatus(g)
-	}, nil)
+
+	err = gui.createConfirmationPanel(v, gui.Tr.SLocalize("Fixup"), message,
+		func(g *gocui.Gui, v *gocui.View) error {
+
+			err := gui.GitCommand.SquashFixupCommit(branch.Name, commit.Sha)
+			if err != nil {
+
+				err = gui.createErrorPanel(err.Error())
+				if err != nil {
+					gui.Log.Errorf("Failed to create error panel at handleCommitFixup: %s\n", err)
+					return err
+				}
+
+				return nil
+			}
+
+			err = gui.refreshCommits()
+			if err != nil {
+				gui.Log.Errorf("Failed to refresh commits at handleCommitFixup: %s\n", err)
+				return err
+			}
+
+			err = gui.refreshStatus()
+			if err != nil {
+				gui.Log.Errorf("Failed to refresh status at handleCommitFixup: %s\n", err)
+				return err
+			}
+
+			return nil
+		}, nil)
+	if err != nil {
+		gui.Log.Errorf("Failed to create confirmation panel at handleCommitFixup: %s\n", err)
+		return err
+	}
+
 	return nil
 }
 
+// handleRenameCommit is called when a user wants to rename a commit.
+// g and v are passed by the gocui library but only v is used.
+// If anything goes wrong it returns an error.
 func (gui *Gui) handleRenameCommit(g *gocui.Gui, v *gocui.View) error {
-	if gui.getItemPosition(gui.getCommitsView(g)) != 0 {
```

Comment:
You know there actually is a reason why we are not passing `v` to this function.

We need to explicitly pass commit panel view because without it, the execution from menu panel of any command involving this function will fail. With `v` it will take ItemPosition from menu panel and this is not desired i guess.

So, pls keep that in mind.

Reply:
Glad you shared your concern, i will change things a bit.